packages/component/src/hooks/internal/useMemoized.ts

```diff
@@ -33,17 +33,30 @@ export default function useMemoized<TFinal, TArgs>(fn: Fn<TArgs, TFinal>, deps:
       nextCacheRef.current = [];
 
       const memoizedFn = (...args) => {
-        const { current: fn } = fnRef;
-        const { current: cache } = cacheRef;
-        const { current: nextCache } = nextCacheRef;
-        const { result } = [...cache, ...nextCache].find(
+        const fn = fnRef.current;
+        const cache = cacheRef.current;
+        const nextCache = nextCacheRef.current;
+
+        const cached = cache.find(
           ({ args: cachedArgs }) =>
-            args.length === cachedArgs.length && args.every((arg, index) => Object.is(arg, cachedArgs[+index]))
-        ) || { result: fn(...args) };
+            // index is guarented to be a number here
+            // eslint-disable-next-line security/detect-object-injection
+            args.length === cachedArgs.length && args.every((arg, index) => Object.is(arg, cachedArgs[index]))
```

Comment:
Could consider this:

```suggestion
            args.length === cachedArgs.length && args.every((arg, index) => Object.is(arg, cachedArgs[+index]))
```

Reply:
The conversion is unnecessary in this case. To me this is more like a tooling issue, so I disabled the specific check.